prover/prover_dal/src/fri_prover_dal.rs

```diff
@@ -568,6 +564,30 @@ impl FriProverDal<'_, '_> {
         .map(|row| row.id as u32)
     }
 
+    pub async fn get_recursion_tip_proof_job_id(
+        &mut self,
+        l1_batch_number: L1BatchNumber,
+    ) -> Option<u32> {
+        sqlx::query!(
+            r#"
+            SELECT
+                id
+            FROM
+                prover_jobs_fri
+            WHERE
+                l1_batch_number = $1
+                AND status = 'successful'
+                AND aggregation_round = $2
+            "#,
+            l1_batch_number.0 as i64,
```

Comment:
Nit: Core DAL [enables](https://github.com/matter-labs/zksync-era/blob/83c40341c092dda71c02fb0cc8c498c5f70c0fe4/core/lib/dal/src/lib.rs#L4) the aforementioned `clippy::cast_lossless` lint to make conversions like these more idiomatic.